flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/operators/aggregate/window/WindowAggOperatorBuilder.java

```diff
@@ -170,4 +202,17 @@ public SlicingWindowAggOperatorBuilder countStarIndex(int indexOfCountStart) {
         }
         return new SlicingWindowOperator<>(windowProcessor);
     }
+
+    @SuppressWarnings("unchecked")
+    private WindowOperatorBase<RowData, ?> buildUnslicingWindowOperator() {
+        final UnsliceWindowAggProcessor windowProcessor =
+                new UnsliceWindowAggProcessor(
+                        (GeneratedNamespaceAggsHandleFunction<TimeWindow>)
+                                generatedAggregateFunction,
+                        (UnsliceAssigner<TimeWindow>) assigner,
+                        accSerializer,
+                        indexOfCountStart,
+                        shiftTimeZone);
+        return new UnslicingWindowOperator<>(windowProcessor);
```

Comment:
Why remove the UnsupportedOperationException?

Reply:
I think it's unnecessary to bind the UnslicingWindowOperator and session window too strongly here.